yt_dlp/extractor/crunchyroll.py

```diff
@@ -714,48 +776,25 @@ class CrunchyrollBetaIE(CrunchyrollBaseIE):
         },
         'params': {'skip_download': 'm3u8'},
         'expected_warnings': ['Unable to download XML']
+    }, {
+        'url': 'https://beta.crunchyroll.com/watch/GY2P1Q98Y/',
+        'only_matching': True,
     }]
 
     def _real_extract(self, url):
-        lang, internal_id, display_id = self._match_valid_url(url).group('lang', 'internal_id', 'id')
-        webpage = self._download_webpage(url, display_id)
-        initial_state = self._parse_json(
-            self._search_regex(r'__INITIAL_STATE__\s*=\s*({.+?})\s*;', webpage, 'initial state'),
-            display_id)
-        episode_data = initial_state['content']['byId'][internal_id]
+        lang, internal_id, display_id = self._match_valid_url(url).group('lang', 'id', 'display_id')
+
         if not self._get_cookies(url).get('etp_rt'):
+            initial_state, app_config = self._get_embedded_json(self._download_webpage(url, display_id), display_id)
+            episode_data = initial_state['content']['byId'][internal_id]
             video_id = episode_data['external_id'].split('.')[1]
             series_id = episode_data['episode_metadata']['series_slug_title']
+            self.to_screen(f'{display_id}: Not logged in. Switching extractors. New url: https://www.crunchyroll.com/{lang}{series_id}/{display_id}-{video_id}')
```

Comment:
```suggestion
```

Reply:
shouldn't some care be taken to ensure the extractors don't keep switching back and forth?

Reply:
Probably. It can't happen now, but future changes to crunchyroll might cause it. I'm not sure how best to actually store the necessary state, though.

Or perhaps it should be done without state? The non-beta extractor could refuse to switch and raise an error instead if not logged in.

Reply:
The standard way to bind a state to a URL is to do `url = smuggle_url(url, state_dict)`. Then the receiving extractor can do `url, smuggled_data = unsmuggle_url(url)`